File: 4242-sum-of-gcd-of-formed-pairs/sum-of-gcd-of-formed-pairs.java
class Solution {
    public long gcd(long a,long b){
        if(b==0)return a;
        return gcd(b,a%b);
    }
    public long gcdSum(int[] nums) {
        int mx= 0;
        ArrayList<Long> f= new ArrayList<>();
        for (int i=0;i<nums.length;i++){
            mx = Math.max(nums[i],mx);
            f.add(gcd(nums[i],mx));
        }
        Collections.sort(f);
        int i=0;
        int j = f.size()-1;
        long sum=0;
        while(i<j){
            sum +=gcd(f.get(i),f.get(j));
            i++;
            j--;
        }
        return sum;
    }
}

File: 4242-sum-of-gcd-of-formed-pairs/sum-of-gcd-of-formed-pairs.py
class Solution:
    def gcdSum(self, nums: list[int]) -> int:
        mx= 0
        f=[]
        for i in range(0,len(nums)):
            mx = max(nums[i],mx)
            f.append(gcd(nums[i],mx))
        f.sort()
        i=0
        j = len(f)-1
        sum=0
        while i<j:
            sum +=gcd(f[i],f[j])
            i+=1
            j-=1
        return sum
        
            
        